File: python-solutions/problem006.py
from time import time

def sum_of_squares(n):
    return n * (n + 1) * (2 * n + 1) / 6

def square_of_sum(n):
    return (n * (n + 1) / 2) ** 2

def compute():
    return square_of_sum(100) - sum_of_squares(100)

if __name__ == "__main__":
    start = time()
    answer = compute()
    end = time() - start

    print(f"\nAnswer for Problem 6: {answer}")
    print( f"Time Taken: {end} seconds\n")

File: golang-solutions/problem006.go
package main

import (
	"fmt"
	"time"
)

func main() {
	start := time.Now()

	answer := squareOfSum(100) - sumOfSquares(100)

	fmt.Printf("\nAnswer to Problem 6: %d\n", answer)
	fmt.Printf("Time Taken: %v\n\n", time.Since(start))
}

func squareOfSum(n int) int {
	return (n * (n + 1) / 2) * (n * (n + 1) / 2)
}

func sumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}
